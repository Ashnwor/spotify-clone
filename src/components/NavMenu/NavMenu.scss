@import '../../styles/color-palette';

.playlist-menu {
  width: 100%;
  padding: 0 24px;
  background: $color-black-2;
}

.menu-title {
  top: 0;
  width: 100%;
  height: 24px;
  flex: none;
  align-self: center;
  order: 0;
  color: $color-gray-3;

  font-family: 'Segoe UI';
  font-size: 12px;
  font-style: normal;
  font-weight: 700;
  letter-spacing: 0.1em;
  line-height: 16px;
}
