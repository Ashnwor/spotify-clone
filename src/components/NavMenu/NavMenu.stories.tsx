import NavMenu from './NavMenu';

export default {
  title: 'Playlist Menu',
};

export const playlistMenu: React.FC = () => <NavMenu />;
