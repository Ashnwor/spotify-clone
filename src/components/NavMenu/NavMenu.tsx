import React from 'react';
import './NavMenu.scss';
import NavMenuItem from '../NavMenuItem/NavMenuItem';
import NavDivider from '../NavDivider/NavDivider';
import { PlaylistHeart, PlaylistPlus } from '../icons';

interface Props {
  className?: string;
}

const NavMenu: React.FC<Props> = ({ className }) => {
  return (
    <div className={`playlist-menu ${className}`}>
      <div className="menu-title">PLAYLISTS</div>
      <NavMenuItem text="Create a playlist" Icon={PlaylistPlus} />
      <NavMenuItem text="Liked Songs" Icon={PlaylistHeart} gradient />
      <NavDivider />
    </div>
  );
};

NavMenu.defaultProps = { className: '' };
export default NavMenu;
