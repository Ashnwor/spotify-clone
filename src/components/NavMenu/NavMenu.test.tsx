import { shallow } from 'enzyme';
import NavMenu from './NavMenu';

it('runs correctly', () => {
  const wrapper = shallow(<NavMenu />);
  expect(wrapper).toMatchSnapshot();
});
