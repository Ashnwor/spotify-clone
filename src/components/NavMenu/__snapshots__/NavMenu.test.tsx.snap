// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runs correctly 1`] = `
<div
  className="playlist-menu "
>
  <div
    className="menu-title"
  >
    PLAYLISTS
  </div>
  <NavMenuItem
    Icon={[Function]}
    text="Create a playlist"
  />
  <NavMenuItem
    Icon={[Function]}
    gradient={true}
    text="Liked Songs"
  />
  <NavDivider />
</div>
`;
